### DISCUSSION TITLE
[Question] Determine available artifact version given the IProject or pom IFile

### DISCUSSION THREAD

**Discussion by BoykoAlex** [eclipse-m2e/m2e-core#1084] [Q&A]
Is there a way with m2e core to compute all versions of an artifact (a dependency artifact) available for a project? Version of an artifact available for the project would reside in an artifact repository specified in the pom or maven central.

I thought I'd go with IMaven.readProject(File, IProgressMonitor) but then i have to deal with `org.apache.maven.project.MavenProject` which isn't made visible... Same goes to `ArtifactRepository` which seem to have the function I'd like to use, namely: `List<String> findVersions(Artifact artifact);`

I found `org.eclipse.m2e.core.ui.internal.search.util.SearchEngine` interface but could find any concrete implementation of it. Besides depending on UI bundles isn't a great idea for me since I'd like to use that code from the Language Server side.

Anything you could recommend or give some pointers?

Thanks in advance!

**Comment by mickaelistria** [comment]
> then i have to deal with org.apache.maven.project.MavenProject which isn't made visible...

I don't understand that part. What do you mean specifically here?

**Reply by BoykoAlex** [reply]
@mickaelistria NVM :-) I had `MavenProject` and other artifacts from `org.apache.maven` unresolved... however, adding a plugin dependency on `org.eclipse.m2e.maven.runtime` makes them available... I've realized this after asking this question. I think i have everything i need to get all available versions for an artifact...
